middleware/activation.py

```
@@ -191,19 +196,38 @@ def _activate_from_scan(toolhead: str, scan: ScanEvent, spool_info: SpoolInfo |
+                "remaining_g": remaining,
+                "spoolman_id": spool_info.spoolman_id if spool_info else None,
+            }
+        if spoolman_activated:
+            logger.info("[afc_stage] Spool staged with Spoolman ID, scanner remains unlocked")
+        else:
+            logger.info("[afc_stage] Tag data cached, waiting for lane load. Scanner remains unlocked")
+
+    elif action == "afc_lane":
         if spoolman_activated:
-            # SET_SPOOL_ID was sent — AFC queries Spoolman for color/material/weight.
             logger.debug(f"AFC lane data via Spoolman (spool_id={spool_info.spoolman_id})")
```

Ruff (0.15.6)

[warning] Logging statement uses f-string

(G004)